### ISSUE TITLE
How weights are used in LightGBM

### ISSUE THREAD

**Issue by JerryX2014** [microsoft/LightGBM#2544]
Hello, I tried to build a model with a binary target and weights by using lightGBM in R and compared it to a XGBoost model. The model is just one tree with subsample = 1 and colsample_bytree = 1. Then I run the lgb.model.dt.tree function to check the model and compared its output to the result from the xgb.model.dt.tree for the XGBoost model. 

Then I found that, for the root node,
1) both lightGBM and XGBoost use the same attribute with the same split point. 
2) The split_gain (from lightGBM) is the same as the Quality (from XGBoost).
3) The internal count from lightGBM is the volume of the data, while the counterpart from XGBoost model is the sum of the weights. 

Based on the definition 3.1 in the lightGBM paper, it seems that the algorithm uses weights from the data to calculate the gradients (the nominator), but uses weights = 1 for the denominator. Is that true? Thanks.

**Comment by guolinke** [comment]
you can check `internal_weight`. 
LightGBM both use count and weight. Count is mainly used for `min_data_per_leaf` constraint. 
The weight is used for split gain calculation and leaf outputs, and the `min_weight_per_leaf` constraint.

**Comment by JerryX2014** [comment]
Thanks Guolin. But I couldn't find internal_weight... I am using lightgbm v2.2.4 in R.

**Comment by guolinke** [comment]
https://github.com/microsoft/LightGBM/search?utf8=%E2%9C%93&q=internal_weight&type=

**Comment by JerryX2014** [comment]
OK, I found the reason

The internal_weight is outputted in the JSON file, but lgb.model.dt.tree function doesn't output internal_weight information. The code below shows it outputs internal_value and internal_count, but not internal_weight.

https://github.com/microsoft/LightGBM/blob/7feaaefb5b730524d12449e48d93a862a668a024/R-package/R/lgb.model.dt.tree.R#L99-L114

I added the definition to the single.tree.parse function and all looks good. Thanks!
